File: tests/fixtures/broken_python/bad_bounds_shape/broken.py
def validate_bounds(x0, lb, ub):
    if len(lb) != len(x0) or len(ub) != len(x0):
        raise ValueError("bounds shape mismatch: expected length 2, got 1")


validate_bounds([0.0, 1.0], [0.0], [2.0, 3.0])

File: tests/fixtures/broken_matlab/bad_bounds_shape/broken.m
% BUG: lower bound vector does not match x0 length.
x0 = [0; 1];
lb = 0;
ub = [2; 3];
if numel(lb) ~= numel(x0) || numel(ub) ~= numel(x0)
    error('Bounds shape mismatch: expected length 2');
end
disp([lb(:), ub(:)]);
